## Importing libraries

In [2]:
# importing relevant libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Reading the data from the imdb datasets

In [3]:
#Loading data on movies info
Movies= pd.read_csv("tmdb.movies.csv")
Movies.head()

Unnamed: 0            genre_ids     id original_language   
0           0      [12, 14, 10751]  12444                en  \
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date   
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19  \
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [5]:
#Summary 
Movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


In [4]:
#Loading data on ratings
ratings =pd.read_csv("title.ratings.csv")
ratings.head()

tconst  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [19]:
#Summary
ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   tconst         73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB


In [5]:
#Loading Budget dat
budget =pd.read_csv("tn.movie_budgets.csv")
budget.head()

id  release_date                                        movie   
0   1  Dec 18, 2009                                       Avatar  \
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [50]:
budget.dtypes

id                    int64
release_date         object
movie                object
production_budget    object
domestic_gross       object
worldwide_gross      object
Total gross          object
dtype: object

In [21]:
#Summary
budget.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [6]:
#Loading  gross data
gross =pd.read_csv("bom.movie_gross.csv")
gross.head()


title studio  domestic_gross   
0                                  Toy Story 3     BV     415000000.0  \
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [16]:
#Summary
gross.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


## Data Analysis

Only a few features and rows from the loadeddatasets that were gathered are important to the procedure. The columns from each dataset that are not necessary will be removed in this step. After then, the remaining datasets will be used to create data visuals


### 1.Dropping Columns

In [7]:
## Selecting necessary columns in the Movies dataset
Movies_new=Movies[["title","popularity"]]
Movies_new

title  popularity
0      Harry Potter and the Deathly Hallows: Part 1      33.533
1                          How to Train Your Dragon      28.734
2                                        Iron Man 2      28.515
3                                         Toy Story      28.005
4                                         Inception      27.920
...                                             ...         ...
26512                         Laboratory Conditions       0.600
26513                               _EXHIBIT_84xxx_       0.600
26514                                  The Last One       0.600
26515                                  Trailer Made       0.600
26516                                    The Church       0.600

[26517 rows x 2 columns]

In [29]:
Movies_new.shape

(26517, 2)

The new Movie dataset has 26517 rows and 2 columns

In [29]:
#Selecting the necessary columns in budget dataset
budget_new =budget[["movie","production_budget", "domestic_gross", "worldwide_gross"]]
budget_new

movie production_budget   
0                                          Avatar         425000000  \
1     Pirates of the Caribbean: On Stranger Tides         410600000   
2                                    Dark Phoenix         350000000   
3                         Avengers: Age of Ultron         330600000   
4               Star Wars Ep. VIII: The Last Jedi         317000000   
...                                           ...               ...   
5777                                       Red 11              7000   
5778                                    Following              6000   
5779                Return to the Land of Wonders              5000   
5780                         A Plague So Pleasant              1400   
5781                            My Date With Drew              1100   

     domestic_gross worldwide_gross  
0         760507625      2776345279  
1         241063875      1045663875  
2          42762350       149762350  
3         459005868      1403013963  
4         620181382      1316721747  
...             ...             ...  
5777              0               0  
5778          48482          240495  
5779           1338            1338  
5780              0               0  
5781         181041          181041  

[5782 rows x 4 columns]

In [43]:
budget_new.shape

(5782, 5)

The new budget dataset has 5782 rows and 5 columns

In [9]:
#Selecting the necessary columns in gross dataset
gross_new =gross[["title", "domestic_gross", "foreign_gross"]]
gross_new

title  domestic_gross   
0                                     Toy Story 3     415000000.0  \
1                      Alice in Wonderland (2010)     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1     296000000.0   
3                                       Inception     292600000.0   
4                             Shrek Forever After     238700000.0   
...                                           ...             ...   
3382                                    The Quake          6200.0   
3383                  Edward II (2018 re-release)          4800.0   
3384                                     El Pacto          2500.0   
3385                                     The Swan          2400.0   
3386                            An Actor Prepares          1700.0   

     foreign_gross  
0        652000000  
1        691300000  
2        664300000  
3        535700000  
4        513900000  
...            ...  
3382           NaN  
3383           NaN  
3384           NaN  
3385           NaN  
3386           NaN  

[3387 rows x 3 columns]

In [79]:
gross_new.shape

(3387, 3)

The new gross dataset has 3387 rows and 3 colums

In [45]:
#Selecting the necessary columns in ratings dataset
ratings_new =ratings[["averagerating", "numvotes"]]
ratings_new

averagerating  numvotes
0                8.3        31
1                8.9       559
2                6.4        20
3                4.2     50352
4                6.5        21
...              ...       ...
73851            8.1        25
73852            7.5        24
73853            4.7        14
73854            7.0         5
73855            6.3       128

[73856 rows x 2 columns]

In [49]:
ratings_new.shape

(73856, 2)

The new ratings dataset has 73856 rows and 2 columns

### 2 Formatting Dataframes

##### 2.1 Budget Dataframe

In [12]:
#Checking if the  new budget dataframe has missing values
budget_new.isna().sum().head()

movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

The dataframe has no missing values

In [26]:
#Removing the special characters ($ and ,)
#regex=True , is used to find all non-word characters, i.e. characters which are not alphabetical or numerical.
budget['domestic_gross'] = budget['domestic_gross'].str.replace('\W', '' ,regex=True)
budget['production_budget'] = budget['production_budget'].str.replace('\W', '' ,regex=True)
budget.head()

id  release_date                                        movie   
0   1  Dec 18, 2009                                       Avatar  \
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0         425000000      760507625      2776345279  
1         410600000      241063875      1045663875  
2         350000000       42762350       149762350  
3         330600000      459005868      1403013963  
4         317000000      620181382      1316721747

The special characters are no longer there

In [37]:
#checking the datatypes of the columns
budget_new.dtypes

movie                object
production_budget    object
domestic_gross       object
worldwide_gross      object
dtype: object

In [ ]:
budget.plot(
   x='production_budget', 
   y='domestic_gross', 
   kind='line'
)

plt.show()

In [ ]:
budget_new['worldwide_gross']=budget_new['worldwide_gross'].astype('int64')
budget_new.dtypes

In [41]:
budget_new.plot.bar(x='domestic_gross', y='worldwide_gross', rot=0);

TypeError: no numeric data to plot

In [ ]:
budget.plot(rot = 60)
plt.xlabel("Index", size = 20)
plt.ylabel("domestic_gross", size = 20)
plt.title("Minimum temperature of Seattle", size = 25)

In [ ]:
gross_new=gross_new.dropna()

In [ ]:
gross_new["domestic_gross"]= gross_new["domestic_gross"].astype(str)

In [ ]:
scatter_plot_title = 'domestic_gross and foreign_gross'
standing_tackle_label = 'domestic_gross'
sliding_tackle_label = 'foreign_gross'

# YOUR CODE HERE
tackle_figure,ax=plt.subplots(figsize=(10,6))
ax.scatter(gross_new["domestic_gross"],gross_new["foreign_gross"])
plt.title(scatter_plot_title)
plt.xlabel(standing_tackle_label)
plt.ylabel(sliding_tackle_label)
plt.show